include/h2o/tunnel.h

```diff
@@ -27,9 +27,53 @@ extern "C" {
 #endif
 
 typedef struct st_h2o_tunnel_t h2o_tunnel_t;
+typedef struct st_h2o_tunnel_endpoint_t h2o_tunnel_endpoint_t;
+typedef void (*h2o_tunnel_endpoint_open_cb)(h2o_tunnel_t *tunnel, h2o_tunnel_endpoint_t *end);
+typedef void (*h2o_tunnel_endpoint_send_cb)(h2o_tunnel_t *tunnel, h2o_tunnel_endpoint_t *end, h2o_iovec_t *bufs, size_t bufcnt,
+                                        int is_final);
+typedef void (*h2o_tunnel_endpoint_on_peer_send_complete_cb)(h2o_tunnel_t *tunnel, h2o_tunnel_endpoint_t *end, h2o_tunnel_endpoint_t *peer);
```

Comment:
Does "peer_send_complete" mean that we are signaling the peer that the send action initiated by the peer has completed? I would appreciate it if you could change the name to better clarify the role.